### ISSUE TITLE
uploaded files are in an endless loop

### ISSUE THREAD

**Issue by deralexxx** [google/timesketch#701]
Uploading a file via API or web ui can cause an endless loop trying to import the file but without any visible progress.

After asking in Slack, @berggren pointed me to look at the celery worker logs at /var/log/celery

The version I used was the latest HEAD version from github within a vagrant instance.

My logs looked like: 

````
[2018-11-07 18:59:33,846: INFO/MainProcess] Connected to redis://127.0.0.1:6379//
[2018-11-07 18:59:33,864: INFO/MainProcess] mingle: searching for neighbors
[2018-11-07 18:59:34,882: INFO/MainProcess] mingle: all alone
[2018-11-07 18:59:34,900: INFO/MainProcess] w1@timesketch-dev ready.
[2018-11-07 19:14:26,621: INFO/MainProcess] Connected to redis://127.0.0.1:6379//
[2018-11-07 19:14:26,631: INFO/MainProcess] mingle: searching for neighbors
[2018-11-07 19:14:27,652: INFO/MainProcess] mingle: all alone
[2018-11-07 19:14:27,657: INFO/MainProcess] w1@timesketch-dev ready.

````

So something was wrong, there where no jobs visible.
After searching in the timesketch github repo I found the following command to start the workers:

````
bash -c "/usr/local/bin/celery -A timesketch.lib.tasks worker --uid nobody --loglevel info"

````

Then I got the workers running and new jobs appeared as planned:

````
[2018-11-09 18:41:42,905: INFO/ForkPoolWorker-2] Task timesketch.lib.tasks.run_csv_jsonl[3618ed49-d28b-402c-ac01-e48ba3282458] succeeded in 0.248744058999s: None
[2018-11-09 18:41:48,311: INFO/ForkPoolWorker-2] Index timeline [sample] to index [1765c1fb5ce54e1b9d24b3a771486f1a] (source: csv)
[2018-11-09 18:41:48,462: INFO/ForkPoolWorker-2] Task timesketch.lib.tasks.run_csv_jsonl[428c3759-30ff-4e7d-a9a8-950f57e5fa83] succeeded in 0.151471524001s: '1765c1fb5ce54e1b9d24b3a771486f1a'
[2018-11-07 18:59:33,846: INFO/MainProcess] Connected to redis://127.0.0.1:6379//
[2018-11-07 18:59:33,864: INFO/MainProcess] mingle: searching for neighbors
[2018-11-07 18:59:34,882: INFO/MainProcess] mingle: all alone
[2018-11-07 18:59:34,900: INFO/MainProcess] w1@timesketch-dev ready.
[2018-11-07 19:14:26,621: INFO/MainProcess] Connected to redis://127.0.0.1:6379//
[2018-11-07 19:14:26,631: INFO/MainProcess] mingle: searching for neighbors
[2018-11-07 19:14:27,652: INFO/MainProcess] mingle: all alone
[2018-11-07 19:14:27,657: INFO/MainProcess] w1@timesketch-dev ready.
[2018-11-09 18:36:57,792: INFO/MainProcess] sync with celery@timesketch-dev
[2018-11-09 18:39:30,985: INFO/MainProcess] sync with celery@timesketch-dev
[2018-11-09 18:41:32,188: INFO/MainProcess] sync with celery@timesketch-dev
[2018-11-09 18:41:42,655: INFO/MainProcess] Received task: timesketch.lib.tasks.run_csv_jsonl[3618ed49-d28b-402c-ac01-e48ba3282458]  
[2018-11-09 18:41:48,310: INFO/MainProcess] Received task: timesketch.lib.tasks.run_csv_jsonl[428c3759-30ff-4e7d-a9a8-950f57e5fa83]  

````

So Imho there should be two tasks now:

- [ ] Fix it in sourcecode to monitor the status of the workers and kind of userfeedback why nothing happens
- [ ] Add it to troubleshooting guide where to look if something is wrong

**Comment by aguilajesus** [comment]
Hey @deralexxx I took a quick look at this today and #663 created some side effect;. indexing is wrapped now with automatic analysis tasks.

if `ENABLE_INDEX_ANALYZERS` is not enable the upload won't be processed. the fastest way to fix this issue it is to set this variable to [True](https://github.com/google/timesketch/blob/master/timesketch.conf#L156)

As a long term fix, we can remove the [check](https://github.com/google/timesketch/blob/master/timesketch/api/v1/resources.py#L1049) to force indexing and let the code do further [checks](https://github.com/google/timesketch/blob/master/timesketch/lib/tasks.py#L110) for this feature. @berggren wdyt?

**Comment by deralexxx** [comment]
Hi @aguilajesus thx for verifying, I tried to set it as suggested:

``
vagrant@timesketch-dev:/etc$ cat  timesketch.conf | grep "ENABLE"
SSO_ENABLED = False
GOOGLE_IAP_ENABLED = False
UPLOAD_ENABLED = True
GRAPH_BACKEND_ENABLED = False
SIMILARITY_EXPERIMENT_ENABLED = False
ENABLE_INDEX_ANALYZERS = True 
ENABLE_SKETCH_ANALYZERS = True 
``

Restarted timesketch:
``
tsctl runserver -h 0.0.0.0
``

And it is still not picking up (I did not manually start celery- should I have done that)

**Comment by aguilajesus** [comment]
keep in mind that tsctl does not start celery by itself; since you are using vagrant, you can check if celery is running with `systemctl status celery.service` or start|restart the workers with `systemctl start|restart celery.service`

let me know if that helps, or you are seeing other errors.

**Comment by deralexxx** [comment]
Ah I see, well then we should document that as well I think

**Comment by aguilajesus** [comment]
indexing and analyzers were fixed in [903023a](https://github.com/google/timesketch/commit/903023ae7fd214a5360f098c1eba187909c8e45a) and merged in #704 

Please, feel free to open issues if you want to pursue the changes/ideas for celery.
